metricbeat/mb/module/wrapper.go

```diff
@@ -236,6 +236,13 @@ func (msw *metricSetWrapper) fetch(reporter reporter) {
 	case mb.ReportingMetricSetV2:
 		reporter.StartFetchTimer()
 		fetcher.Fetch(reporter.V2())
+	case mb.ReportingMetricSetV2Error:
```

Comment:
If this is going to be the recommended interface, should we change the metricset template to use it too?

Reply:
Good question. I would first try it out and then switch it to the default. At the same time we probably will forget with this approach :-(

Reply:
Yeah, we can add it later, maybe we can create an issue as follow up to decide and change defaults in metricset generators.

Reply:
done: https://github.com/elastic/beats/issues/11079